### DISCUSSION TITLE
请在这个频道发布自己的部署成功经历

### DISCUSSION THREAD

**Discussion by oddfar** [oddfar/campus-imaotai#45] [分享部署教程]
使用群辉、威联通、openwrt等，可以在下方分享自己部署成功的经历，帮助其他人。

**Comment by zhao316678171** [comment]
在群辉7.2系统上兜兜转转最终架设成功了，分享下经验，以供大家避坑：
1、前期基本上按照《威联通上部署分享》这个帖子的内容修改后来架设，群主的源码放在docker下面自建文件夹，用项目新增选择目录，用修改后的yml文件部署，过程中可能报错，最常见的就是缺少某些目录，直接过去新建就行，一路新建容器内会新增4个容器。
![image](https://github.com/oddfar/campus-imaotai/assets/63034357/1c7b5447-87dc-4fbb-90b4-6ad96f6e6db8)
2、关于user: 0:0和user：xxx:xxx这些问题仔细看《威联通上部署分享》里面也有怎么把属于当前账号的xxx：xxx搞出来；
3、最开始我部署后，四个容器redis可以跑起来，mysql死活跑不起来，后来没办法用群辉虚拟机装了linux系统跑起来先用了一周，群辉自带的docker里面项目和容器都没删，找了一个晚上研究下，看着报错尝试修改，最终都跑了起来，下面分享几个注意地方。
4、端口的问题，特别nginx的80端口，最好改下，我改成180，后面群辉会提示让装web station的套件，提前改了防止后面冲突；
5、关于自动创建的mysql容器映射文件夹，看报错提示“什么什么没权限”，我改了conf.d成conf，原理不明白就是报错招不到conf.d。再细看映射文件夹，前面自建的没有这个文件，只有conf，于是改了，后来改完跑起来mysql这个容器了。
![image](https://github.com/oddfar/campus-imaotai/assets/63034357/e1b0c5d4-a7d4-4ddf-8adc-b794deebfa9d)
6、前面redis跑起来了，mysql也跑起来了，server直接跟就跑起来了，最后nginx又是提示报错，看了报错原因，提示nginx.conf，最后一行之前根据《威联通上部署分享》最后一行我家的user：root报错，直接过去删掉，启动nginx就跑起来了。最终四个容器全跑起来了。
7、进主页ip+180没反应进不去，跑到docker项目那里点项目提示要安装web station套间，于是跟着装了，后面再用ip+180端口直接就可以进了。
8、总结就是根据《威联通上部署分享》增加该增加的权限问题，部署完成容器四个容器成功后再逐个启动四个容器，那个报错再解决那个容器问题。
9、最后，如果还是不行，也是我架设成功之前最后一个思路就是把之前虚拟机部署成功生成的文件夹和数据下载上传过来群辉，直接覆盖掉docker下面自建的文件夹，当然你也可以直接这么干，再根据上面思路做调整去尝试。
10、总之，在折腾的道路上不断折腾，不为别的就为了折腾，可能有不对的地方或者其他问题欢迎下面留言，大家一起折腾。
